### DISCUSSION TITLE
Ability to disable media keys

### DISCUSSION THREAD

**Discussion by Wielding** [jenius-apps/ambie#220] [Ideas]
I often have ambie running in the background all day and occasionally start up other apps that use the play/pause controls without stopping ambie.  It would be great to have the option to prevent ambie from reacting to the play/pause controls so that the app I want to control receives the events instead.  Usually this is for videos related to work that I need to constantly start/stop for reference.  If ambie is running it will often get the play/pause event instead of the video and then I have to juggle the apps to get them both playing again.  I did my own local build of ambie and just commented out the code so I could get this result but it would be nice to have it as an option in the official app since I don't want to keep updating my quick fix with new build, plus I am a paid user so I don't get the premium sounds with my local build.

**Comment by dpaulino** [comment]
yep I can make an option for this

**Reply by Wielding** [reply]
Very cool.  Thanks.

**Reply by dpaulino** [reply]
Now available in 2.0.18

**Reply by Wielding** [reply]
Thanks.  I notice that now the sounds don't play while Ambie is minimized.   Is that a side effect of not integrating with the Windows Media controls?  My local build with the play/pause handler commented out works fine when minimized.

**Reply by dpaulino** [reply]
oh no.. yes it seems that fully disabling the integration also disabled background audio. I'll send another update that just disables the play/pause handlers. Thanks for spotting that!! I should have tested it more.

**Reply by dpaulino** [reply]
@Wielding to be clear, was it these that you commented out?
![image](https://user-images.githubusercontent.com/7649589/150899317-4ea36b00-6fad-4cde-83f5-a6d4f29290d5.png)

**Comment by Wielding** [comment]
No, I commented out these lines in MixMediaPlayerService.cs to just disable the Pause which also breaks the GUI button but I did not use that anyway since I just start the sound and minimize the app.   When I don't want the sound I just close the app.  I did not dig very deep looking for a solution.

![image](https://user-images.githubusercontent.com/5303456/150899579-5f6dcaf7-a5bf-4c54-80e6-6097e2df94b2.png)

**Reply by dpaulino** [reply]
Thanks. I changed something different, but I think it will have me same effect as your commented out code. It's available in 2.0.19 now

**Reply by Wielding** [reply]
I took a look and now it plays in the background and the media keys are ignored which is great, but the media keys presses don't pass through to other apps.  I did some testing with my local build and noticed that it works intermittently.  I have not been in a situation recently to actively use that functionality so I did not realize it was not consistent. 

I am not sure how UWP apps intercept the media key presses.  In my C++ apps where I use low level keyboard hooks I have to return a result to say if I have processed it or not so it can be processed by the next hook.  I have no idea if that translates in any way to how a C# UWP app gets the media key presses.

I am not at all familiar with UWP apps or XAML so I am looking around the code to see what I can pick up on.  I only write win32 Windows apps with C++ so most of this is new to me.  My feeling is that the UWP framework is going to prevent this from working the way I would like due to its limitations of interacting with the system for security.  Please don't spend a lot of time on this.  It would be a nice feature but if it is disruptive, I think it should be left where it is.  I wrote a little win32 app that generates brown noise when I am working with other media players that can fill the spot when I am in that situation.  Of course, it does not sound as nice as Ambie.

**Comment by Wielding** [comment]
I did try a few other locations that did not work and I don't remember where they were.
